midterm-2023-11-07/code/5263-angela-kostadinova/script.ts

```diff
@@ -0,0 +1,106 @@
+interface Movie {
+
+id: number;
+title: string;
+director: string;
+year: number;
+genre: string;
+plot: string;
+cast: string[];
+oscars: number;
+}
+async function LoadingTheData() {
+    const dataUri = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-07/data/movies.json";
+    const response = await fetch(dataUri);
+
+    if (!response.ok) {
+        throw new Error("Ooops...Sorry, I cannot take requested data!");
+    }
+
+    const data = await response.json();
+    return data;
+}
+document.addEventListener("DOMContentLoaded", siteCode);
+function CastListGenerate(cast: string[]) {
+    const ul = document.createElement("ul");
+    for (const actor of cast) {
+        const li = document.createElement("li");
+        li.innerHTML = actor;
+        ul.appendChild(li);
+    }
+    return ul;
+}
+function DisplayAllMovies(movies: Movie[]) {
+    const container = document.getElementById("movie-container")!;
+    container.innerHTML = "";
+
+    for (const movie of movies) {
+        const movieTable = generateMovieTable(movie);
+        container.appendChild(movieTable);
+    }
+}
+function generateMovieTable(movie: Movie) {
+    const table = document.createElement("div");
+    table.classList.add("movie-table");
+    const movieData = [
+    movie.id,
+    movie.title,
+    movie.director,
+    movie.year,
+    movie.genre,
+    movie.plot,
+
+        CastListGenerate(movie.cast),
+        movie.oscars
+    ];
+    for (const data of movieData) {
+        const movieDataElement = document.createElement("div");
+        movieDataElement.classList.add("movie-data");
+        movieDataElement.innerHTML = data.toString();
+        table.appendChild(movieDataElement);
+    }
+    return table;
+}
+
+let movies: Movie[] = [];
+async function siteCode() {
+    const data = await LoadingTheData();
+    movies = data;
+
+    DisplayAllMovies(movies);
+
+}
+const applyFilter = () => {
+    const directorElement = document.getElementById("director-filter") as HTMLSelectElement;
+    const director = directorElement.value;
+
+    const yearElement = document.getElementById("year-filter") as HTMLInputElement;
+    const year = yearElement.value;
+    let filteredMovies = movies; 
+    if (director !== "all") {
```

Comment:
Yes, assuming the director and year are correct, this will filter out the needed movies +10